tests/fixtures/dbt_integration_project.py

```diff
@@ -0,0 +1,101 @@
+import pytest
+
+dbt_integration_project__my_macros_sql = """
+
+
+{% macro do_something(foo, bar) %}
+
+    select
+        '{{ foo }}'::text as foo,
+        '{{ bar }}'::text as bar
+
+{% endmacro %}
+
+"""
+
+
+dbt_integration_project__incremental_sql = """
+
+-- TODO : add dist/sort keys
+{{
+    config(
+        materialized = 'incremental',
+        unique_key   = 'id',
+    )
+}}
+
+
+select * from {{ this.schema }}.seed
+
+{% if is_incremental() %}
+    where updated_at > (select max(updated_at) from {{ this }})
+{% endif %}
+"""
+
+
+dbt_integration_project__schema_yml = """
+version: 2
+models:
+- name: table_model
+  columns:
+  - name: id
+    tests:
+    - unique
+"""
+
+
+dbt_integration_project__table_model_sql = """
+
+-- TODO : add dist/sort keys
+{{
+    config(
+        materialized = 'table',
+    )
+}}
+
+select * from {{ this.schema }}.seed
+"""
+
+dbt_integration_project__view_model_sql = """
+{{
+    config(
+        materialized = 'view',
+    )
+}}
+
+select * from {{ this.schema }}.seed
+"""
+
+
+dbt_integration_project__dbt_project_yml = """
+name: dbt_integration_project
```

Comment:
instead of strings, why not have all of these as actual files that are read in?

Reply:
Great question! Defining actual files, in subdirectories modelled on dbt project structures, is how our old integration testing framework used to work.

This was a choice that the Core Eng team made earlier this year, when developing our new `pytest`-based framework for functional testing. The biggest reasons:
- Easier to see/edit/update all project components if defined in the same file
- Easier to reuse/update/modify these if they're Python objects (strings), rather than files
- Abstract away the file system (to some extent), since it's annoying to handle the differences across operating systems

There's one downside: It's slightly harder to take this project and `dbt run` with it locally. There are some utilities in the testing framework that make it pretty easy to convert a project like this one into a real local file system. (I end up just dropping a breakpoint, finding the temp file location, and copying the directory into a new location—but I'm sure there's a more ergonomic way.)

Reply:
If you run the functional tests with "-s" it will print out the temporary directory where the project is located.